framework/Optimizers/GradientDescent.py

```diff
@@ -22,6 +22,7 @@
 #End compatibility block for Python 3----------------------------------------------------------------
 
 #External Modules------------------------------------------------------------------------------------
+import copy
```

Comment:
leftover?

Reply:
copy appears to be used (line 534) I think